numcosmo/data/nc_data_cluster_wl.c

```diff
@@ -464,6 +464,8 @@ nc_data_cluster_wl_integ (NcmIntegralND *intnd, NcmVector *x, guint dim, guint n
   NcDataClusterWLInt *lh_int = NC_DATA_CLUSTER_WL_INT (intnd);
   guint i;
 
+  nc_galaxy_sd_shape_integ_prep (lh_int->data.s_dist, lh_int->data.cosmo, lh_int->data.dp, lh_int->data.smd, lh_int->data.z_cluster, lh_int->data.r);
```

Comment:
By calling this function here you are basically preparing every time you compute a new point of the parametric space! This would only work if you use a vectorized method below. This should be moved below at line 589. You also do not need to create and destroy the object (line 581) for each galaxy. You create it once and just change the values and the preparation before calling the integral.

Reply:
Fixed both!